dao-generator/src/main/java/com/exactpro/cradle/cassandra/dao/retry/AbstractRetryDao.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.cradle.cassandra.dao.retry;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.oss.driver.api.mapper.MapperContext;
+
+public abstract class AbstractRetryDao {
+
+    public static final String MIN_TIMEOUT_KEY = "minTimeout";
+    public static final String MAX_TIMEOUT_KEY = "maxTimeout";
+    public static final String COUNT_ATTEMPTS_KEY = "countAttempts";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRetryDao.class);
+
+    private final long minTimeout;
+    private final long countAttempts;
+    private final long h;
+
+    public AbstractRetryDao(MapperContext context) {
+        Map<Object, Object> customState = context.getCustomState();
+
+        if (customState.get(MIN_TIMEOUT_KEY) != null) {
+            minTimeout = ((Number) customState.get(MIN_TIMEOUT_KEY)).longValue();
+        } else {
+            minTimeout = 1000L;
+        }
+
+        long maxTimeout;
+
+        if (customState.get(MAX_TIMEOUT_KEY) != null) {
+            maxTimeout = ((Number) customState.get(MAX_TIMEOUT_KEY)).longValue();
+        } else {
+            maxTimeout = 60000L;
+        }
+
+        if (customState.get(COUNT_ATTEMPTS_KEY) != null) {
+            countAttempts = ((Number) customState.get(COUNT_ATTEMPTS_KEY)).longValue();
+        } else {
+            countAttempts = 5L;
+        }
```

Comment:
These three cases (minTimeout, maxTimeout, countAttempts) can be got using one method.